fullscreen/api/element-request-fullscreen-same-element.html

```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE html>
+<title>Element#requestFullscreen() on the current fullscreen element</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<div id="log"></div>
+<script>
+promise_test(async (t) => {
+    t.add_cleanup(() => { document.exitFullscreen() });
+
+    // Use the body element as the fullscreen element, because the second
+    // test_driver.bless() call needs to insert a button inside of it, which
+    // can't be clicked if another element is already fullscreen.
+    const target = document.body;
+
+    // First enter fullscreen.
+    await test_driver.bless("fullscreen", () => target.requestFullscreen());
+    assert_equals(document.fullscreenElement, target, "fullscreen element after first request");
+
+    // Now request fullscreen again, which should be a no-op.
+    await test_driver.bless("fullscreen", () => target.requestFullscreen());
+    assert_equals(document.fullscreenElement, target, "fullscreen element after second request");
+});
```

Comment:
```suggestion
promise_test(async (t) => {
    t.add_cleanup(() => { document.exitFullscreen() });

    const div = document.querySelector("div");

    await test_driver.bless("fullscreen");
    
    // Request fullscreen twice in a row. The first request should consume the
    // user activation and succeed, and the second request should fail because
    // of the user activation requirement.
    const p1 = div.requestFullscreen();
    const p2 = promise_rejects_js(t, TypeError, div.requestFullscreen());
    await Promise.all([p1, p2]);
});
```